src/cpp/lpnamer/model/ChronicleMapReader.h

```diff
@@ -0,0 +1,32 @@
+//
+// Created by marechaljas on 28/04/2022.
+//
+
+#ifndef ANTARESXPANSION_TESTS_CPP_LP_NAMER_CHRONICLEMAPREADERTEST_CPP_CHRONICLEMAPREADER_H_
+#define ANTARESXPANSION_TESTS_CPP_LP_NAMER_CHRONICLEMAPREADERTEST_CPP_CHRONICLEMAPREADER_H_
+#include <map>
+#include <sstream>
+
+class ScenarioToChronicleReader {
+ private:
+  mutable std::map<unsigned, unsigned > chronicle_map;
+
+  void ignoreFirstLine(std::stringstream& ss) const;
+
+  void AssignChronicleValueToMCYear(std::stringstream& ss,
+                                    unsigned int montecarlo_year) const;
+  void AssignChronicleValuesToMCYears(std::stringstream& ss) const;
+
+ public:
+  [[nodiscard]] std::map<unsigned int, unsigned int> read(std::string const& input) const;
+
+  [[nodiscard]] std::map<unsigned int, unsigned int> read(std::stringstream& input) const;
+
+  [[nodiscard]] std::map<unsigned int, unsigned int> read(
+      const std::ifstream& input) const;
```

Comment:
You have 3 versions of `read`. I suggest that you remove the `std::stringstream&` and the `const std::ifstream&` versions, and create a generic `std::istream&` instead.

Example
```cpp
#include <iostream>
#include <sstream>
#include <fstream>
#include <cassert>

int read(std::istream& is) {
  int out;
  is >> out;
  return out;
}

int main() {
  {
    std::string s("1");
    std::stringstream ss(s);
    auto a = read(ss);
    assert(a == 1);
  }

  {
    std::ifstream f("test.txt"); // 2
    auto a = read(f);
    assert(a == 2);
  }
}
```

Reply:
Fixed with only 2 version. One with streams, one with string